library/src/main/scala/g8.scala

```diff
@@ -462,35 +474,45 @@ object G8 {
     }
   }
 
-  def readProps(stm: java.io.InputStream): G8.OrderedProperties = {
-    val p = new LinkedListProperties
-    p.load(stm)
-    stm.close()
-    (OrderedProperties.empty /: p.keyList) { (l, k) =>
-      l :+ (k -> p.getProperty(k))
+  def readProps(stm: InputStream): G8.OrderedProperties = {
+    // Overrides java.util.Properties to return properties back in the order they were specified
+    val properties = new Properties {
+      val parameters = mutable.MutableList.empty[(String, String)]
+
+      override def put(key: Object, value: Object) = {
+        val k = key.toString
+        val v = value.toString
+
+        parameters += k -> v
+        super.put(k, v)
+      }
+    }
+
+    try {
+      properties.load(stm)
+      properties.parameters.toList
+    } finally {
+      stm.close()
     }
   }
+
+  private val truthy = Set("y", "yes", "true")
+  private val falsy = Set("n", "no", "false")
```

Comment:
i don't think it is possible. templates return strings, not booleans. there is no way in ST itself to convert from strings to booleans; if there were, #157 wouldn't be needed to begin with. see the [discussion on conditionals](https://github.com/antlr/stringtemplate4/blob/master/doc/templates.md#discussion-2), in particular:

> Strict separation of model and view requires that expressions cannot test attribute values such as name=="parrt"

i don't think this issue can be solved with ST trickery alone, without changes to scala code. i don't have lots of experience with ST, but that is my understanding. please correct me if i'm wrong.

honestly, i'm very partial now to the new proposed solution: keep the string parameter verbatim and create a companion boolean. ending booleans with `?` is a convention in some programming languages, ruby i think. it is not something totally absurd :-)